### DISCUSSION TITLE
[Q&A] How to identify which molecule is organic?

### DISCUSSION THREAD

**Discussion by IchiruTake** [rdkit/rdkit#5314] [General] [answered]
Given only the general molecule only, is there any particular wide-range definition of classifying the molecule to be organic or not?
Some online definition attempted different methods and used exception to handle but it is relatively complex. 
For example, some agreed CCl4 is organic and some don't. 

Is the basic attempt for classification is to check the atom type then filter out the exception?

**Comment by greglandrum** [comment, marked as answer]
@IchiruTake I think the hardest part of this is defining what you mean by "organic", the rest is probably a relatively straightforward function.

Here's a simple definition which illustrates an approach to solve this once you figure out a definition you're happy with:
```
simpleOrganicAtomQuery = Chem.MolFromSmarts('[!$([#1,#5,#6,#7,#8,#9,#15,#16,#17,#35,#53])]')
simpleOrganicBondQuery = Chem.MolFromSmarts('[#6]-,=,#,:[#6]')
hasCHQuery = Chem.MolFromSmarts('[C!H0]')
def simple_is_organic(mol):
     return (not mol.HasSubstructMatch(simpleOrganicAtomQuery)) and mol.HasSubstructMatch(hasCHQuery) and mol.HasSubstructMatch(simpleOrganicBondQuery)
```

**Reply by IchiruTake** [reply]
Thank you @greglandrum, the approach is relatively straightforward and satisfied my need. According to Wikipedia, we can check the availability of C-C bond as well, which can test the mol such as `N#CC#N`

**Reply by greglandrum** [reply]
@IchiruTake I wasn't paying close enough attention and had used the word "aromatic" in my sample code instead of "organic". I just fixed that.
The pattern `simpleOrganicBondQuery` already checks for the existence of any kind of CC bond

**Reply by IchiruTake** [reply]
Yes, I am aware that issue but the definition of organic molecules should have extra attention on the real-world chemistry reaction (this applied for `H-C#N`). But I think the solution is good enough. We may need to have extra exception to filter out the accepted molecule but inorganic one. Thank you

**Reply by IchiruTake** [reply]
> @IchiruTake I wasn't paying close enough attention and had used the word "aromatic" in my sample code instead of "organic". I just fixed that. The pattern `simpleOrganicBondQuery` already checks for the existence of any kind of CC bond

Don't worry about that
